diffrax/_solver/align.py

```diff
@@ -0,0 +1,164 @@
+import equinox as eqx
+import jax.numpy as jnp
+import jax.tree_util as jtu
+from equinox.internal import ω
+from jaxtyping import Array, PyTree
+
+from .._custom_types import (
+    AbstractSpaceTimeLevyArea,
+    RealScalarLike,
+)
+from .._local_interpolation import LocalLinearInterpolation
+from .._term import _LangevinArgs, LangevinTerm, LangevinTuple, LangevinX
+from .langevin_srk import _AbstractCoeffs, _SolverState, AbstractLangevinSRK
+
+
+# UBU evaluates at l = (3 -sqrt(3))/6, at r = (3 + sqrt(3))/6 and at 1,
+# so we need 3 versions of each coefficient
+
+
+class _ALIGNCoeffs(_AbstractCoeffs):
+    @property
+    def dtype(self):
+        return jtu.tree_leaves(self.beta)[0].dtype
+
+    beta: PyTree[Array]
+    a1: PyTree[Array]
+    b1: PyTree[Array]
+    aa: PyTree[Array]
+    chh: PyTree[Array]
+
+
+_ErrorEstimate = LangevinTuple
+
+
+class ALIGN(AbstractLangevinSRK[_ALIGNCoeffs, _ErrorEstimate]):
+    r"""The Adaptive Langevin via Interpolated Gradients and Noise method
+    designed by James Foster. Only works for Underdamped Langevin Diffusion
+    of the form
+
+    $$d x_t = v_t dt$$
+
+    $$d v_t = - gamma v_t dt - u ∇f(x_t) dt + (2gammau)^(1/2) dW_t$$
+
+    where $v$ is the velocity, $f$ is the potential, $gamma$ is the friction, and
+    $W$ is a Brownian motion.
+    """
+
+    term_structure = LangevinTerm
+    interpolation_cls = LocalLinearInterpolation
+    taylor_threshold: RealScalarLike = eqx.field(static=True)
+    _coeffs_structure = jtu.tree_structure(
+        _ALIGNCoeffs(
+            beta=jnp.array(0.0),
+            a1=jnp.array(0.0),
+            b1=jnp.array(0.0),
+            aa=jnp.array(0.0),
+            chh=jnp.array(0.0),
+        )
+    )
+
+    @property
+    def minimal_levy_area(self):
+        return AbstractSpaceTimeLevyArea
+
+    def __init__(self, taylor_threshold: RealScalarLike = 0.0):
+        r"""**Arguments:**
+
+        - `taylor_threshold`: If the product `h*gamma` is less than this, then
+        the Taylor expansion will be used to compute the coefficients.
+        Otherwise they will be computed directly. When using float32, the
+        empirically optimal value is 0.1, and for float64 about 0.01.
+        """
+        self.taylor_threshold = taylor_threshold
```

Comment:
Maybe let's set this to `0.1` by default?

Also it looks like you could assign this default value directly to the field and skip having an `__init__` method here?

Reply:
Yeah, I originally had it at 0.1 by default but then Trevor asked me to change it to 0. But I agree with you (and hopefully Trevor won't notice :)).